src/common/docker-compose/index.ts

```diff
@@ -253,19 +255,29 @@ export class DockerComposeUtils {
             service_to.labels.push(`traefik.port=${gateway_port}`);
           }
 
-          const host = interface_name === '@' ? 'arc.localhost' : `${interface_name}.arc.localhost`;
+          // eslint-disable-next-line prefer-const
+          let [interface_subdomain, component_interface_name] = interface_name.split('->'); // remove extra part of subdomain used for deduping here
+
+          const host = interface_subdomain === '@' ? 'arc.localhost' : `${interface_subdomain}.arc.localhost`;
 
-          if (interface_name === '@') {
+          if (interface_subdomain === '@') {
             // @ is an invalid service name for traefik
-            interface_name = '__at__';
+            interface_subdomain = '__at__';
+          }
+
+          const interfaces_node = graph.getUpstreamNodes(node_to).find(n => n instanceof InterfacesNode);
+          const component_interface = (interfaces_node as InterfacesNode).component_interfaces[component_interface_name];
+          if (interfaces_node && component_interface?.ingress?.path) {
+            service_to.labels.push(`traefik.http.routers.${interface_subdomain}.rule=Host(\`${host}\`) && Path(\`${component_interface.ingress.path}.*\`)`);
```

Comment:
Worried this might be a bad assumption:
`${component_interface.ingress.path}.*`

This is essentially traefik's path prefix:
https://doc.traefik.io/traefik/routing/routers/#rule

Lets poll Brahm/Dan on default behavior.

Wondering if we should require that someone specify the `.*`

Reply:
I guess I'd prefer that as long as we make sure that people understand what that means under the hood